src/megaapi_impl.cpp

```diff
@@ -5124,14 +5114,12 @@ void MegaApiImpl::init(MegaApi *api, const char *appKey, MegaGfxProcessor* proce
 
     if (basePath)
     {
-        string sBasePath = basePath;
-        int lastIndex = int(sBasePath.size() - 1);
-        if (sBasePath[lastIndex] != '/' && sBasePath[lastIndex] != '\\')
+        LocalPath lp = LocalPath::fromPath(basePath, *fsAccess);
+        if (!lp.endsInSeparator(fsAccess->localseparator))
```

Comment:
Not important: I believe appendWidthSeparator already does this check

Reply:
I think you're right.  Updated